tests/test_cleaneval.py

```diff
@@ -0,0 +1,486 @@
+        if tag == "delete":
+            fp += text_l
+            tag_fp += text_tags
+        elif tag == "insert":
+            fn += gold_l
+            tag_fn += gold_tags
+        else:
+            tp += text_l
+            tag_tp += text_tags
+            assert text_l == gold_l
+            assert text_tags == gold_tags
+    result = __evaluate(tp, fp, fn, tag_tp, tag_fp, tag_fn)
+    return result
+
+def __evaluate(tp, fp, fn, tag_tp, tag_fp, tag_fn):
+    n_text = tp + fp if tp + fp > 0 else 1                                                                                                              
+    n_gold = tp + fn if tp + fn > 0 else 1
+    precision = float(tp) / n_text
+    recall = float(tp) / n_gold
+    precision_plus_recall = precision + recall if precision + recall > 0 else 1
+    f_score = 2 * precision * recall / precision_plus_recall
+                                                                                                                                                        
+    tags_text = tag_tp + tag_fp if tag_tp + tag_fp > 0 else 1
+    tags_gold = tag_tp + tag_fn if tag_tp + tag_fn > 0 else 1
+    tag_precision = float(tag_tp) / tags_text
+    tag_recall = float(tag_tp) / tags_gold
+    precision_plus_recall = tag_precision + tag_recall if tag_precision + tag_recall > 0 else 1
+    tag_f_score = 2 * tag_precision * tag_recall / precision_plus_recall
+                                                                                                                                                        
+    return (100 * f_score, 100 * precision, 100 * recall, 100 * tag_f_score, 100 * tag_precision, 100 * tag_recall, tp, fp, fn, tag_tp, tag_fp, tag_fn)
+
+def write_alignment(diff, filename):
+	fh = file(filename, "w")
+	for tag, text_seg, gold_seg in diff:
+		if tag == "delete":
+			print("<" * 40, "(false positive)", file=fh)
+			print(" ".join(text_seg), file=fh)
+		if tag == "insert":
+			print(">" * 40, "(false negative)", file=fh)
+			print(" ".join(gold_seg), file=fh)
+		if tag == "equal":
+			print("=" * 40, file=fh)
+			print(" ".join(gold_seg), file=fh)
+	fh.close()
+
+def do_cleaneval(gold_html, cleaned_html):
+    '''
+    Arguments:
+        gold_html       String  the gold standard html
+        cleaned_html    String  the cleaned html returned from a library
+    Return:
+        eval_list       tuple   representing score on cleaned text evaluation comparing with the gold standard
+
+    >>> do_cleaneval(' <h> header <p> content', '<h> header <p> content') 
+    (100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 5, 0, 0, 2, 0, 0)
+    '''
+    cleaned_html = normalize(cleaned_html)
+    text_words = re_WS.split(cleaned_html)
+    gold_words = re_WS.split(gold_html)
+    alignment = SequenceMatcher(None, text_words, gold_words)
+    diff = make_diff(alignment, text_words, gold_words)
+    eval_list = evaluate(diff)
+    return eval_list
+
+def do_jaccard(gold_html, cleaned_html):
```

Comment:
The `do_jaccard` and `do_jaccard_multiset` functions are basically the same thing, except that the former uses `set` and the latter uses `Multiset`; it'd be a little bit nicer if you combined them into a single function in order to reduce code duplication; there's lots of ways to do that, and we can talk about some of them if you can't think of a good way yourself

Reply:
at the very least there needs to be a comment in the docstrings explaining the code duplication so that someone editing the code knows they need to modify both functions if they modify one of them